### DISCUSSION TITLE
Illegal Adress Problem

### DISCUSSION THREAD

**Discussion by MGGEL** [pymodbus-dev/pymodbus#1934] [Help] [answered]
Hi,
I get an "IllegalAddress" Message when I try to read a register.
But the register is correct and it exists is in the documentation:

![image](https://github.com/pymodbus-dev/pymodbus/assets/47329154/0d9f9815-c006-4800-9550-9b180c3c11ed)

Source:
`response = client.read_input_registers(40003, 1, 0)`

Logging:

```
DEBUG:pymodbus.logging:Connection to Modbus server established. Socket ('192.168.188.129', 51214)
DEBUG:pymodbus.logging:Current transaction state - IDLE
DEBUG:pymodbus.logging:Running transaction 1
DEBUG:pymodbus.logging:SEND: 0x0 0x1 0x0 0x0 0x0 0x6 0x0 0x4 0x9c 0x43 0x0 0x1
DEBUG:pymodbus.logging:New Transaction state "SENDING"
DEBUG:pymodbus.logging:Changing transaction state from "SENDING" to "WAITING FOR REPLY"
DEBUG:pymodbus.logging:Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
DEBUG:pymodbus.logging:RECV: 0x0 0x1 0x0 0x0 0x0 0x3 0x0 0x84 0x2
DEBUG:pymodbus.logging:Processing: 0x0 0x1 0x0 0x0 0x0 0x3 0x0 0x84 0x2
DEBUG:pymodbus.logging:Factory Response[132]
DEBUG:pymodbus.logging:Adding transaction 1
DEBUG:pymodbus.logging:Getting transaction 1
DEBUG:pymodbus.logging:Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
Fehler beim Lesen des Registers: Exception Response(132, 4, IllegalAddress)
```

If i try to read any register above 9999 the message IllegalAddress occurs. 
Might that a problem of the bit order or something like that?

**Comment by janiversen** [comment, marked as answer]
pymodbus convert to big endian automatically as you can see in the debug log. Big endian is defined by the modbus protocol as the only way to transmit data.

The exception is a response from your device "illegal address" so you might want to read the device manual.
